zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -522,23 +525,24 @@ private void tryRejectingIfUserCommand(final String errorMessage) {
     final var rejectionReason = errorMessage != null ? errorMessage : "";
     final ProcessingResultBuilder processingResultBuilder =
         new BufferedProcessingResultBuilder(logStreamWriter::canWriteEvents);
-    // reset value to minimize any potential error loop that can be caused by writing the full
-    // record.
-    typedCommand.getValue().reset();
-    final var rejectionMetadata =
+    final var errorRecord = new ErrorRecord();
+    errorRecord.initErrorRecord(new RuntimeException(rejectionReason), currentRecord.getPosition());
```

Comment:
:upside_down_face: Very soft suggestion, I would prefer a specific exception so it's immediately obvious what it is, but you can still find out via stack trace (and technically no one is reading this record ;)). Plus, it's not the hot path, so we don't need to think about a stackless exception anyway. But a `CommandRejectionException` might not be a bad thing in the protocol in general :shrug:

Reply:
Replaced RuntimeException with `CommandRejectionException`.